vignettes/design-principles.Rmd

```diff
@@ -0,0 +1,48 @@
+---
+title: "Design Principles for {packagetemplate}"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Design Principles for {packagetemplate}}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette outlines the design decisions that have been taken during the development of the {packagetemplate} R package, and provides some of the reasoning, and possible pros and cons of each decision.
+
+This document is primarily intended to be read by those interested in understanding the code within the package and for potential package contributors.
+
+<!-- Text in angular brackets (<>) is to guide what to include for each section please delete the guidelines and replace for your package.
+
+None of the sections are required, feel free to remove any sections not relevant to your package or add new sections if helpful. -->
+
+## Scope
+
+< Outline the aims of the package, potentially mention some of the key exported functions, and maybe how it links with other R packages. It is also possible to mention certain aspects that fall outside of the package's scope. >
+
+## Input/Output/Interoperability
+
+< Describe the data structures (i.e. vectors, `<data.frames>` or classes) that are given as input to the key functions and what data structures the functions return. The design decisions around these I/O choices could also mention how it enhances interoperability with other R packages or pipelines (e.g. with `%>%`). >
+
+
+## Design decisions
+
+< A list of bullet points each explaining a design decision and its reasoning. >
+
+## Dependencies
```

Comment:
Would it be worth adding more guidance on which dependencies should be listed here? Some strategies could be:

- just the key dependencies
- a general policy for dependency (e.g., "this package is expected to be used in tidyverse pipelines and as such, we consider all tidyverse packages to be good dependency candidates")
- just the dependencies that would stand out or surprise a new contributor. Contributors will likely not be surprised we use testthat but may be surprised about others

Reply:
Thanks for the suggestions, I have used each of these to provide more explanation on possible topics to discuss in the dependencies section ([64117aa](https://github.com/epiverse-trace/packagetemplate/pull/97/commits/64117aa4b4b4ed5ad27f43e6a4dcc7d433beff97)).